### PR TITLE
Revive the "send email to new users" toggle for the user form

### LINKED ISSUE #11172: Missing setting to send welcome mail when creating a user
I am missing the setting to disable/enable to send a welcome mail when creating a user.

Currently, an email is always sent, whether you want it or not.

In Nextcloud 13:

unbenannt

In Nextcloud 14:

unbenannt1

Please bring back the setting that the admin can disable/enable the welcome mail.

### Server configuration

**Operating system**: Debian 9

**Web server:** Apache2.4

**Database:** MariaDB

**PHP version:** PHP7.0

**Nextcloud version:** 14.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 13.0.6 and fresh install.

**Where did you install Nextcloud from:**

### PR DESCRIPTION
**Pull request by ChristophWurst**
Fixes https://github.com/nextcloud/server/issues/11172

This is the minimalist version, so we can easily backport it. Someone can later integrate this better into the new form.

Todo
- [x] Add back toggle in bottom left
- [x] Send preference to server with user data
- [x] Only send email when admin wishes so
- [x] Persist toggle value

### CODE REVIEW

File: apps/settings/src/views/Users.vue [hunk 1]
```diff
@@ -276,6 +285,19 @@ export default {
...
+		sendWelcomeMail: {
+			get() {
+				return this.settings.newUserSendEmail
+			},
+			set(value) {
+				this.$store.commit('setServerData', {
+					...this.settings,
+					newUserSendEmail: value
+				})
+				axios.post(generateUrl(`/settings/users/preferences/newUser.sendEmail`), { value: value ? 'yes' : 'no' })
```
Comment (skjnldsv) [inline apps/settings/src/views/Users.vue]: Can lead to race condition, right?

Comment (ChristophWurst) [reply apps/settings/src/views/Users.vue]: Which ones?

Comment (skjnldsv) [reply apps/settings/src/views/Users.vue]: If you quickly toggle the checkbox and the first is a little longer to resolve than the other?

Comment (ChristophWurst) [reply apps/settings/src/views/Users.vue]: Hm. Yeah. But what is the solution? If it's debounced then only the first trigger has an effect. We could disable it

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 2]
```diff
@@ -335,7 +335,7 @@ public function addUser(string $userid,
 			}
 
 			// Send new user mail only if a mail is set
-			if ($email !== '') {
+			if ($email !== '' && $this->config->getAppValue('core', 'newUser.sendEmail', 'no') === 'yes') {
```
Comment (rullzer) [inline apps/provisioning_api/lib/Controller/UsersController.php]: should this not default to yes?

**Review by rullzer — CHANGES_REQUESTED** [review]

File: apps/settings/src/views/Users.vue (hunk 1 above)
Comment (skjnldsv) [reply apps/settings/src/views/Users.vue]: omfg

**Review by skjnldsv — APPROVED** [review]

**Review by rullzer — APPROVED** [review]
Seems to do the trick!

File: apps/settings/lib/Controller/UsersController.php [hunk 3]
```diff
@@ -238,10 +241,28 @@ public function usersList() {
...
 		return new TemplateResponse('settings', 'settings-vue', ['serverData' => $serverData]);
 	}
 
+	/**
+	 * @param string $key
+	 * @param string $value
+	 *
+	 * @return JSONResponse
+	 */
+	public function setPreference(string $key, string $value): JSONResponse {
```
Comment (nickvergessen) [inline apps/settings/lib/Controller/UsersController.php]: We have an ocs api to set app configs. It even comes with a nice JS api for it

**Comment by ChristophWurst** [comment]
We don't need this in 16.

**Comment by skjnldsv** [comment]
ok :)

**Pull request merged**